diff --git a/node_modules/@hmscore/react-native-hms-analytics/.DS_Store b/node_modules/@hmscore/react-native-hms-analytics/.DS_Store
new file mode 100644
index 0000000..bda48b4
Binary files /dev/null and b/node_modules/@hmscore/react-native-hms-analytics/.DS_Store differ
diff --git a/node_modules/@hmscore/react-native-hms-analytics/ios/HmsAnalytics.swift b/node_modules/@hmscore/react-native-hms-analytics/ios/HmsAnalytics.swift
index e7c2378..7086ac9 100644
--- a/node_modules/@hmscore/react-native-hms-analytics/ios/HmsAnalytics.swift
+++ b/node_modules/@hmscore/react-native-hms-analytics/ios/HmsAnalytics.swift
@@ -107,10 +107,13 @@ class HMSAnalytics: NSObject, Handling{
     ///   - reject:  Refers to promise value, in the failure scenarario, exception is returned.
     /// - Returns: Void
     @objc func addDefaultEventParams(_ params: NSDictionary, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
-        Log.debug(#function) {
-            viewModel.addDefaultEventParams(params: params)
-            handle(resolve: resolve)
+        if let paramsDict = params as? Dictionary<String, Any> {
+            Log.debug(#function) {
+                viewModel.addDefaultEventParams(paramsDict)
+                handle(resolve: resolve)
+            }
         }
+
     }
 
     /// Enable AB Testing. Predefined or custom user attributes are supported.
diff --git a/node_modules/@hmscore/react-native-hms-analytics/ios/HmsAnalytics.xcodeproj/xcuserdata/pf00142106.xcuserdatad/xcschemes/xcschememanagement.plist b/node_modules/@hmscore/react-native-hms-analytics/ios/HmsAnalytics.xcodeproj/xcuserdata/pf00142106.xcuserdatad/xcschemes/xcschememanagement.plist
new file mode 100644
index 0000000..b500a17
--- /dev/null
+++ b/node_modules/@hmscore/react-native-hms-analytics/ios/HmsAnalytics.xcodeproj/xcuserdata/pf00142106.xcuserdatad/xcschemes/xcschememanagement.plist
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>SchemeUserState</key>
+	<dict>
+		<key>HMSAnalytics.xcscheme_^#shared#^_</key>
+		<dict>
+			<key>orderHint</key>
+			<integer>0</integer>
+		</dict>
+	</dict>
+</dict>
+</plist>
diff --git a/node_modules/@hmscore/react-native-hms-analytics/ios/HmsAnalytics.xcworkspace/xcuserdata/pf00142106.xcuserdatad/UserInterfaceState.xcuserstate b/node_modules/@hmscore/react-native-hms-analytics/ios/HmsAnalytics.xcworkspace/xcuserdata/pf00142106.xcuserdatad/UserInterfaceState.xcuserstate
new file mode 100644
index 0000000..b3e0d69
Binary files /dev/null and b/node_modules/@hmscore/react-native-hms-analytics/ios/HmsAnalytics.xcworkspace/xcuserdata/pf00142106.xcuserdatad/UserInterfaceState.xcuserstate differ
